### ISSUE TITLE
ResourceServerTokenServicesConfiguration should pick up UserDetailsService beans

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security-oauth2-boot#49]
A user can wire a custom `UserDetailsService` to a Resource Server by exposing it through a token store:

```java
@Bean
    public TokenStore tokenStore() {
        DefaultUserAuthenticationConverter userAuthConverter = new DefaultUserAuthenticationConverter();
        userAuthConverter.setUserDetailsService(customUserDetailsService);
        DefaultAccessTokenConverter accessTokenConverter = new DefaultAccessTokenConverter();
        accessTokenConverter.setUserTokenConverter(userAuthConverter);
        return new JwkTokenStore(jwkSetUri, accessTokenConverter);
    }
```

This could be simplified, through by picking up any `UserDetailsService` `@Bean` from the application context, removing the need to define a custom token store.

**Comment by jzheaux** [comment]
@berndgoetz, thanks again for the suggestion!

This is now merged to master via  67497e5

**Comment by berndgoetz** [comment]
Thanks for picking it up so quickly!

**Comment by berndgoetz** [comment]
I've done a test with 2.1.0.RC1, I'm getting the following startup error:

The bean 'scopedTarget.oauth2ClientContext', defined in class path resource [org/springframework/boot/autoconfigure/security/oauth2/client/OAuth2RestOperationsConfiguration$SessionScopedConfiguration$ClientContextConfiguration.class], could not be registered. A bean with that name has already been defined in org.springframework.security.oauth2.config.annotation.web.configuration.OAuth2ClientConfiguration$OAuth2ClientContextConfiguration and overriding is disabled.

**Comment by berndgoetz** [comment]
I've updated the dependencies in https://github.com/berndgoetz/spring-sec-oauth2-rs-user-details-service, I still had the old version on the classpath. Now it works.

**Comment by berndgoetz** [comment]
@jzheaux can you please check the build.gradle file in the test project? Is the dependency set reasonable / best practice from your point of view? Thanks.

**Comment by jzheaux** [comment]
That's right @berndgoetz. That dependency is managed through [spring-cloud-security](https://github.com/spring-cloud/spring-cloud-security/blob/master/spring-cloud-security-dependencies/pom.xml) which hasn't released yet. In the meantime, specifying the version as you have done will work.

**Comment by jzheaux** [comment]
@berndgoetz I apologize, but we've had to revert this enhancement. I failed to consider the fact that since a `UserDetailsService` is always exposed by default in Spring Security, there isn't a good way to differentiate when a user has specified a `UserDetailService` or not.

I'll keep my mind open for other possible avenues, but for the time being, you'll need to go back to having a custom token store when you upgrade.
